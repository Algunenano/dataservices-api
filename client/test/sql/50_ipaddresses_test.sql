\set VERBOSITY terse
-- Add to the search path the schema
SET search_path TO public,cartodb,cdb_dataservices_client;

-- Mock the server functions
CREATE OR REPLACE FUNCTION cdb_dataservices_server.cdb_geocode_ipaddress_point(username text, orgname text, ip_address text)
RETURNS Geometry AS $$
BEGIN
  RAISE NOTICE 'cdb_dataservices_server.cdb_geocode_ipaddress_point invoked with params (%, %, %)', username, orgname, ip_address;
  RETURN NULL;
END;
$$ LANGUAGE 'plpgsql';


-- -- Exercise the public and the proxied function

-- No permissions granted
SELECT cdb_geocode_ipaddress_point('8.8.8.8');

-- Grant other permissions but geocoding
SELECT CDB_Conf_SetConf('api_keys_postgres', '{"permissions": ["routing", "isolines"]}');
SELECT cdb_geocode_ipaddress_point('8.8.8.8');

-- Grant geocoding permissions
SELECT CDB_Conf_SetConf('api_keys_postgres', '{"permissions": ["geocoding"]}');
SELECT cdb_geocode_ipaddress_point('8.8.8.8');

-- Remove permissions
SELECT CDB_Conf_RemoveConf('api_keys_postgres');
