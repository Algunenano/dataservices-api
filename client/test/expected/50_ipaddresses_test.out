\set VERBOSITY terse
-- Add to the search path the schema
SET search_path TO public,cartodb,cdb_dataservices_client;
-- Mock the server functions
CREATE OR REPLACE FUNCTION cdb_dataservices_server.cdb_geocode_ipaddress_point(username text, orgname text, appname text, ip_address text)
RETURNS Geometry AS $$
BEGIN
  RAISE NOTICE 'cdb_dataservices_server.cdb_geocode_ipaddress_point invoked with params (%, %, %, %)', username, orgname, appname, ip_address;
  RETURN NULL;
END;
$$ LANGUAGE 'plpgsql';
-- -- Exercise the public and the proxied function
-- No permissions granted
SELECT cdb_geocode_ipaddress_point('8.8.8.8');
ERROR:  Geocoding is not allowed
-- Grant other permissions but geocoding
SELECT CDB_Conf_SetConf('api_keys_postgres', '{"application": "testing_app", "permissions": ["routing", "isolines"]}');
 cdb_conf_setconf 
------------------
 
(1 row)

SELECT cdb_geocode_ipaddress_point('8.8.8.8');
ERROR:  Geocoding is not allowed
-- Grant geocoding permissions
SELECT CDB_Conf_SetConf('api_keys_postgres', '{"application": "testing_app", "permissions": ["geocoding"]}');
 cdb_conf_setconf 
------------------
 
(1 row)

SELECT cdb_geocode_ipaddress_point('8.8.8.8');
NOTICE:  cdb_dataservices_client._cdb_geocode_ipaddress_point(4): [contrib_regression] REMOTE NOTICE: cdb_dataservices_server.cdb_geocode_ipaddress_point invoked with params (test_user, <NULL>, "testing_app", 8.8.8.8)
 cdb_geocode_ipaddress_point 
-----------------------------
 
(1 row)

-- Remove permissions
SELECT CDB_Conf_RemoveConf('api_keys_postgres');
 cdb_conf_removeconf 
---------------------
 
(1 row)

